server/src/testFixtures/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java

```diff
@@ -272,4 +272,22 @@ protected void awaitMasterFinishRepoOperations() throws Exception {
         });
     }
 
+    @SuppressWarnings("unchecked")
+    protected SnapshotInfo snapshotInfo(String repo, String snapshot) throws Exception {
+        execute(
+            "SELECT id, concrete_indices, started, finished, failures from sys.snapshots WHERE repository = ? AND name = ?",
+            new Object[]{repo, snapshot}
+        );
+        return new SnapshotInfo(
+            new SnapshotId(snapshot, (String) response.rows()[0][0]),
+            (List<String>) response.rows()[0][1],
+            (Long) response.rows()[0][2],
+            null, // Dummy value, since it's not used in tests and not exposed in sys.snapshots.
+            (Long) response.rows()[0][3],
+            1, // Dummy value, since it's not used in tests and not exposed in sys.snapshots.
+            (List<SnapshotShardFailure>) response.rows()[0][4],
+            true // Dummy value, since it's not used in tests and not exposed in sys.snapshots.
+        );
```

Comment:
Returned SnapshotInfo is used in getIndexShardSnapShotStates -> snapshotShards and some fields are unused neither here in tests nor in production code.

Is it okay to keep it or it can backfire and we should add "reason", "totalShards" to the sys.snapshots?
Basically only GetSnapshotRequest is capable of returning "reason" and "totalShards". I think it's fine to not expose it as it's not used anywhere.

Reply:
I think we should followup here and expose `reason` and `total_shards` at `sys.snapshots`.